bee-ledger/src/workers/consensus/worker.rs

```diff
@@ -0,0 +1,328 @@
+impl<N: Node> Worker<N> for ConsensusWorker
+where
+    N::Backend: StorageBackend,
+{
+    type Config = (SnapshotConfig, PruningConfig);
+    type Error = Error;
+
+    fn dependencies() -> &'static [TypeId] {
+        vec![TypeId::of::<TangleWorker>(), TypeId::of::<SnapshotWorker>()].leak()
+    }
+
+    async fn start(node: &mut N, config: Self::Config) -> Result<Self, Self::Error> {
+        let (snapshot_config, pruning_config) = config;
+        let (tx, rx) = mpsc::unbounded_channel();
+        let tangle = node.resource::<MsTangle<N::Backend>>();
+        let storage = node.storage();
+        let bus = node.bus();
+
+        validate_ledger_state(&*storage).await?;
+
+        let depth = if snapshot_config.depth() < SOLID_ENTRY_POINT_THRESHOLD_FUTURE {
+            warn!(
+                "Configuration value for \"depth\" is too low ({}), value changed to {}.",
+                snapshot_config.depth(),
+                SOLID_ENTRY_POINT_THRESHOLD_FUTURE
+            );
+            SOLID_ENTRY_POINT_THRESHOLD_FUTURE
+        } else {
+            snapshot_config.depth()
+        };
+        let delay_min = snapshot_config.depth() + SOLID_ENTRY_POINT_THRESHOLD_PAST + PRUNING_THRESHOLD + 1;
+        let delay = if pruning_config.delay() < delay_min {
+            warn!(
+                "Configuration value for \"delay\" is too low ({}), value changed to {}.",
+                pruning_config.delay(),
+                delay_min
+            );
+            delay_min
+        } else {
+            pruning_config.delay()
+        };
+
+        // Unwrap is fine because ledger index was already in storage or just added by the snapshot worker.
+        let mut ledger_index = storage::fetch_ledger_index(&*storage).await?.unwrap();
+        let mut receipt_migrated_at = MilestoneIndex(0);
+
+        node.spawn::<Self, _, _>(|shutdown| async move {
+            info!("Running.");
+
+            let mut receiver = ShutdownStream::new(shutdown, UnboundedReceiverStream::new(rx));
+
+            while let Some(ConsensusWorkerEvent(message_id)) = receiver.next().await {
+                if let Err(e) = confirm::<N>(
+                    &tangle,
+                    &storage,
+                    &bus,
+                    message_id,
+                    &mut ledger_index,
+                    &mut receipt_migrated_at,
+                )
+                .await
+                {
+                    error!("Confirmation error on {}: {}.", message_id, e);
+                    panic!("Aborting due to unexpected ledger error.");
+                }
+
+                if !tangle.is_confirmed() {
+                    continue;
+                }
+
+                if should_snapshot(&tangle, MilestoneIndex(*ledger_index), depth, &snapshot_config) {
+                    // if let Err(e) = snapshot(snapshot_config.path(), event.index - depth) {
+                    //     error!("Failed to create snapshot: {:?}.", e);
+                    // }
```

Comment:
Looks like this is TODO going by the content of `pruning/mod.rs`, might be worth marking it.